mods/default/nodes.lua

```diff
@@ -1422,115 +1410,200 @@ local function has_locked_chest_privilege(meta, player)
 	return true
 end
 
-minetest.register_node("default:chest", {
-	description = "Chest",
-	tiles = {"default_chest_top.png", "default_chest_top.png", "default_chest_side.png",
-		"default_chest_side.png", "default_chest_side.png", "default_chest_front.png"},
-	paramtype2 = "facedir",
-	groups = {choppy = 2, oddly_breakable_by_hand = 2},
-	legacy_facedir_simple = true,
-	is_ground_content = false,
-	sounds = default.node_sound_wood_defaults(),
+local function chest_lid_obstructed(pos)
+	local above = { x = pos.x, y = pos.y + 1, z = pos.z }
+	local def = minetest.registered_nodes[minetest.get_node(above).name]
+	-- allow ladders, signs, wallmounted things and torches to not obstruct
+	if def.drawtype == "airlike" or
+			def.drawtype == "signlike" or
+			def.drawtype == "torchlike" or
+			(def.drawtype == "nodebox" and def.paramtype2 == "wallmounted") then
+		return false
+	end
+	return true
+end
 
-	on_construct = function(pos)
+local _open = {}
+
+minetest.register_on_player_receive_fields(function(player, formname, fields)
+	if formname ~= "default:chest" then
+		return
+	end
+	if not fields.quit then
+		return
+	end
+	local pn = player:get_player_name()
+
+	local pos = _open[pn].pos
+	local sound = _open[pn].sound
+	local swap = _open[pn].swap
+	local node = minetest.get_node(pos)
+
+	_open[pn] = nil
+	for k, v in pairs(_open) do
+		if v.pos.x == pos.x and v.pos.y == pos.y and v.pos.z == pos.z then
+			return true
+		end
+	end
+	minetest.swap_node(pos, { name = "default:" .. swap, param2 = node.param2 })
+	minetest.sound_play({ name = sound, gain = 0.3, pos = pos, max_hear_distance = 10})
+	return true
+end)
+
+function default.register_chest(name, def)
+	def.drawtype = "mesh"
+	def.visual = "mesh"
+	def.paramtype = "light"
+	def.paramtype2 = "facedir"
+	def.legacy_facedir_simple = true
+	def.is_ground_content = false
+
+	if def.protected then
+		def.on_construct = function(pos)
+			local meta = minetest.get_meta(pos)
+			meta:set_string("infotext", "Locked Chest")
```

Comment:
Looks good. But. The inside of the chest *isn't* black.

https://www.youtube.com/watch?v=JHfmoG4MUIQ

I've already had user controlled animated nodes. Tsk. Tsk.

Secondly, this could be done with a single mesh.

Not to mention /clearobjects does like to delete things. You don't seem to
replace it once deleted. (Unlidded chests look wrong entirely.)

https://github.com/Jordach/big_freaking_dig/blob/master/mods/deco/init.lua#L133
, (You need one line of node meta for lid state, can be stored between
server restarts, etc)

https://github.com/Jordach/big_freaking_dig/blob/master/mods/deco/init.lua#L940
, (complete with yaw -> degree rotations in place, see mesh .blend - I also
don't see node rotation),

https://github.com/Jordach/big_freaking_dig/blob/master/mods/deco/init.lua#L1082
, (Helper to set the animation frames correctly.).


You're doing it wrong, sort of. Never sort of re-invent the wheel with
existing worlds with chests, and rotations.